docs/resources/task.md

```diff
@@ -279,25 +281,86 @@ spec:
 
 ## Permission Set
 
+### Generic
+
 A set of "permissions" assigned to the `task` instance, format depends on the cloud provider
 
-#### Amazon Web Services
+### Cloud-specific
+
+#### Kubernetes
+
+The name of a service account in the current namespace.
+
+### Amazon Web Services
 
 An [instance profile `arn`](https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/iam-roles-for-amazon-ec2.html), e.g.:
 `permission_set = "arn:aws:iam:1234567890:instance-profile/rolename"`
 
-#### Google Cloud Platform
+### Google Cloud Platform
 
 A service account email and a [list of scopes](https://cloud.google.com/sdk/gcloud/reference/alpha/compute/instances/set-scopes#--scopes), e.g.:
 `permission_set = "sa-name@project_id.iam.gserviceaccount.com,scopes=storage-rw"`
 
-#### Microsoft Azure
+### Microsoft Azure
 A comma-separated list of [user-assigned identity](https://docs.microsoft.com/en-us/azure/active-directory/managed-identities-azure-resources/overview) ARM resource ids, e.g.:
 `permission_set = "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.ManagedIdentity/userAssignedIdentities/{identityName}"`
 
+## Pre-allocated blob container
+
+### Generic
+
+To use a pre-allocated container for storing task data, specify the `container` key in the `storage` section
+of the config:
+
+```hcl
+resource "iterative_task" "example" {
+  (...)
+  storage {
+    container = "container-name/path/path"
+  }
+  (...)
+}
+```
+
+The container name may include a path component, in this case the specified subdirectory will be used
+to store task execution results. Otherwise, a subdirectory will be created with a name matchin the
+task's randomly generated id.
+
+If the container name is suffixed with a forward slash, (`container-name/`), the root of the container
+will be used for storage.
+
+### Cloud-specific
+
+#### Amazon Web Services
+
+The container name is the name of the S3 container. It should be in the same region as the task deployment.
+
+#### Google Cloud Platform
+
+The container name is the name of the google cloud storage container.
+
 #### Kubernetes
 
-The name of a service account in the current namespace.
+The container name is the name of a predefined persistent volume claim.
+
+#### Microsoft Azure
+
+To use a pre-allocated azure blob container, the storage account name and access key need to be specified in
+the `storage` section:
+
+```
+resource "iterative_task" "example" {
+    (...)
+    storage {
+      container = "container-name"
+      container_opts = {
+        account = "storage-account-name"
+        key = "storage-account-key"
+      }
+    }
+    (...)
+}
```

Comment:
Is this still accurate after https://github.com/iterative/terraform-provider-iterative/pull/687#pullrequestreview-1135497855?

Reply:
Yes, as I mentioned, to list and inspect storage accounts, we need to specify a resource group they belong to. So we'd be pass two items of information either way.

Reply:
I was misled because none of these options were present on https://github.com/iterative/terraform-provider-iterative/pull/687#discussion_r994705175 🤔

Reply:
sorry about that - the azure example was a bad paste, updated it

Reply:
I'm still tempted to propose `resource-group/storage-account/container/path` as a shorthand for `rclone` connection strings, and retrieve keys automatically, but that's perhaps taking things a bit too far in terms of vertical whitespace quota economy.